src/view/screens/SavedFeeds.tsx

```diff
@@ -43,150 +51,178 @@ const HITSLOP_BOTTOM = {
-                />
-              ))
-            )
+            preferences?.feeds?.pinned?.map(uri => (
+              <ListItem key={uri} feedUri={uri} isPinned />
+            ))
+          )
+        ) : (
+          <ActivityIndicator style={{marginTop: 20}} />
+        )}
+        <View style={[pal.text, pal.border, styles.title]}>
+          <Text type="title" style={pal.text}>
+            Saved Feeds
+          </Text>
+        </View>
+        {preferences?.feeds ? (
+          !preferences.feeds.unpinned.length ? (
+            <View
+              style={[
+                pal.border,
+                isMobile && s.flex1,
+                pal.viewLight,
+                styles.empty,
+              ]}>
+              <Text type="lg" style={[pal.text]}>
+                You don't have any saved feeds.
+              </Text>
+            </View>
           ) : (
-            <ActivityIndicator style={{marginTop: 20}} />
-          )}
+            preferences.feeds.unpinned.map(uri => (
+              <ListItem key={uri} feedUri={uri} isPinned={false} />
+            ))
+          )
+        ) : (
+          <ActivityIndicator style={{marginTop: 20}} />
+        )}
 
-          <View style={styles.footerText}>
-            <Text type="sm" style={pal.textLight}>
-              Feeds are custom algorithms that users build with a little coding
-              expertise.{' '}
-              <TextLink
-                type="sm"
-                style={pal.link}
-                href="https://github.com/bluesky-social/feed-generator"
-                text="See this guide"
-              />{' '}
-              for more information.
-            </Text>
-          </View>
-          <View style={{height: 100}} />
-        </ScrollView>
-      </CenteredView>
-    )
-  }),
-)
+        <View style={styles.footerText}>
+          <Text type="sm" style={pal.textLight}>
+            Feeds are custom algorithms that users build with a little coding
+            expertise.{' '}
+            <TextLink
+              type="sm"
+              style={pal.link}
+              href="https://github.com/bluesky-social/feed-generator"
+              text="See this guide"
+            />{' '}
+            for more information.
+          </Text>
+        </View>
+        <View style={{height: 100}} />
+      </ScrollView>
+    </CenteredView>
+  )
+})
 
 const ListItem = observer(function ListItemImpl({
-  savedFeeds,
-  item,
+  feedUri,
+  isPinned,
 }: {
-  savedFeeds: SavedFeedsModel
-  item: FeedSourceModel
+  feedUri: string // uri
+  isPinned: boolean
 }) {
   const pal = usePalette('default')
-  const isPinned = item.isPinned
+  const queryClient = useQueryClient()
+  const {isPending: isPinPending, mutateAsync: pinFeed} = usePinFeedMutation()
+  const {isPending: isUnpinPending, mutateAsync: unpinFeed} =
+    useUnpinFeedMutation()
+  const {isPending: isMovePending, mutateAsync: setSavedFeeds} =
+    useSetSaveFeedsMutation()
```

Comment:
No optimistic updates here bc I don't have a shared context to use `variables` and I didn't want to complicate things with the query cache. We can migrate to that if we need to, but for now this works well. Pending states disable buttons while async req is pending.